docs/content/examples/data_scientists.md

```diff
@@ -1,180 +1,153 @@
+        to subscribe to the topic the publisher is pushing articles to
+        """
+        self.topic = topic
+        self.ensign = Ensign(
+            client_id=client_id,
+            client_secret=client_secret
+        )
+        self.NER = spacy.load('en_core_web_sm')
 ```
 
-Then you can add posts with
+The next step is to add a `subscribe` method to access the topic stream:
 
-```bash
-$ baleen posts:add https://www.news-source-of-your-choice.com/link-to-article
-```
+```python
+    async def subscribe(self):
+       """
+       Subscribe to the article and parse the events.
+       """
+       tid = await self.ensign.topic_id(self.topic)
 
-Baleen has an Ensign `Publisher` that emits new events to a topic stream (let's call it `"baleen-docs"`) every time a new article is ingested. We should first check that we can access that particular Ensign stream (*Note: make sure you [create an Ensign `client`]({{< ref "/getting-started/ensign#create-a-client" >}}) first!*):
-
-```golang
-import (
-    // ...
-    ensign "github.com/rotationalio/go-ensign"
-)
-
-// This is the nickname of the topic, which gets mapped to an ID
-// that actually gets used by Ensign to identify the stream
-const Baleen = "baleen-docs"
-
-func main() {
-    // ...
-
-	// Check to see if topic exists and create it if not
-	exists, err := client.TopicExists(context.Background(), Baleen)
-	if err != nil {
-		panic(fmt.Errorf("unable to check topic existence: %s", err))
-	}
-
-	var topicID string
-	if !exists {
-		if topicID, err = client.CreateTopic(context.Background(), Baleen); err != nil {
-			panic(fmt.Errorf("unable to create topic: %s", err))
-		}
-	} else {
-		topics, err := client.ListTopics(context.Background())
-		if err != nil {
-			panic(fmt.Errorf("unable to retrieve project topics: %s", err))
-		}
-
-		for _, topic := range topics {
-			if topic.Name == Baleen {
-				var topicULID ulid.ULID
-				if err = topicULID.UnmarshalBinary(topic.Id); err != nil {
-					panic(fmt.Errorf("unable to retrieve requested topic: %s", err))
-				}
-				topicID = topicULID.String()
-			}
-		}
-	}
-
-    // ...
-}
+       async for sub_event in self.ensign.subscribe(tid):
+           print("here's where we'll put the NLP magic!")
 ```
 
-We can write a `Subscriber` to connect to the Baleen topic feed in order to tap into the `baleen-docs` stream:
-
-```golang
-import (
-    // ...
-    ensign "github.com/rotationalio/go-ensign"
-)
-
-func main() {
-    // ...
+And another method to run the `subscribe` method in a continuous loop:
 
-    var sub ensign.Subscriber
-
-	// Create a downstream consumer for the event stream
-	sub, err = client.Subscribe(context.Background(), topicID)
-	if err != nil {
-		panic(fmt.Errorf("could not create subscriber: %s", err))
-	}
-	defer sub.Close()
-
-    // ...
-}
+```python
+    def run(self):
+        """
+        Run the subscriber forever.
+        """
+        asyncio.get_event_loop().run_until_complete(self.subscribe())
```

Comment:
In the next release of PyEnsign this will be:

```python
asyncio.run(self.subscribe())
```

Should we just update that here now?

Reply:
Thanks for the heads up! I'm a little worried that if people see an old example and aren't able to get it to run they'll quit too early. It's getting hard to keep the docs synchronized with all the changes by myself, so I've reassigned the stories to you that update the data playground and ensign-examples to the new pyensign API.